ospd_openvas/preferencehandler.py

```diff
@@ -0,0 +1,610 @@
+        ] and isinstance(vt_param_value, str):
+            return None
+        elif param_type == 'checkbox' and (
+            vt_param_value == '0' or vt_param_value == '1'
+        ):
+            return None
+        elif param_type == 'file':
+            try:
+                b64decode(vt_param_value.encode())
+            except (binascii.Error, AttributeError, TypeError):
+                return 1
+            return None
+        elif param_type == 'integer':
+            try:
+                int(vt_param_value)
+            except ValueError:
+                return 1
+            return None
+
+        return 1
+
+    def process_vts(
+        self, vts: Dict[str, Dict[str, str]], vts_cache: Dict[str, Dict],
+    ) -> Tuple[List[str], Dict[str, str]]:
+        """ Add single VTs and their parameters. """
+        vts_list = []
+        vts_params = {}
+        vtgroups = vts.pop('vt_groups')
+
+        if vtgroups:
+            vts_list = self.get_vts_in_groups(vtgroups, vts_cache)
+
+        for vtid, vt_params in vts.items():
+            if vtid not in vts_cache:
+                logger.warning(
+                    'The VT %s was not found and it will not be loaded.', vtid
+                )
+                continue
+
+            vts_list.append(vtid)
+            for vt_param_id, vt_param_value in vt_params.items():
+                param_type = self.get_vt_param_type(
+                    vtid, vt_param_id, vts_cache
+                )
+                param_name = self.get_vt_param_name(
+                    vtid, vt_param_id, vts_cache
+                )
+
+                if not param_type or not param_name:
+                    logger.debug(
+                        'Missing type or name for VT parameter %s of %s. '
+                        'It could not be loaded.',
+                        vt_param_id,
+                        vtid,
+                    )
+                    continue
+
+                if vt_param_id == '0':
+                    type_aux = 'integer'
+                else:
+                    type_aux = param_type
+
+                if self.check_param_type(vt_param_value, type_aux):
+                    logger.debug(
+                        'The VT parameter %s for %s could not be loaded. '
+                        'Expected %s type for parameter value %s',
+                        vt_param_id,
+                        vtid,
+                        type_aux,
+                        str(vt_param_value),
+                    )
+                    continue
+
+                if type_aux == 'checkbox':
+                    vt_param_value = _from_bool_to_str(int(vt_param_value))
+
+                vts_params[
+                    "{0}:{1}:{2}:{3}".format(
+                        vtid, vt_param_id, param_type, param_name
+                    )
+                ] = str(vt_param_value)
+
+        return vts_list, vts_params
+
+    def set_plugins(self, vts_cache: Dict[str, Dict],) -> bool:
```

Comment:
Just a thought that comes to my mind may vts_cache change during the lifetime of the PreferenceHandler class? If not it could be a passed to the constructur instead of passing it to several methods.